### ISSUE TITLE
Expected FAILED revolution

### ISSUE THREAD

**Issue by okt-kostik** [oktetlabs/test-environment#54]
In the current world if TE has a TRC a bug associated with the test it's "failed as expected"

I think it's wrong:

 - it's not conclusive that if I have a bug I'm happy with it
 - and it does not allow me to mark test with a bug that is to be investigated, say:
     - we have a session with 1000 failures
     - we want to quickly assign bugs to engineers
     - later on we might decide some bugs are OK

Quoting the TRC from public netdriver repo:

```
 601         <results tags="xilinx_efct" key="NO-TSO" notes="TCP segmentation offload will not be implemented: XN-200494-PD-1D/KD-070">
 602           <result value="SKIPPED">
 603             <verdict>Feature 'tx-tcp-segmentation' is disabled and cannot be modified</verdict>
 604           </result>
 605         </results>
```

Two approaches:

## XML-only

 - ok=true should be specified near key
 - ok=true MUST require explicit key specificication

Default behaviour - `ok=false`

## UI based

This implies that we keep the current behaviour in TRC, but on UI (bublik) level we add a table with KEYs in each bublik project. Bublik users effectively decide on key by key basis which bug is OK and which is not.

Default will be sorted project-wide.

Note, that this way we can't have KEY-XXX ok for one set of tags and NOK for another. But may be it's OK.

@ol-andrewr thoughts?

**Comment by ol-andrewr** [comment]
Since OK/NOK is release-specific I think we should not overload TRC DB (XML). It is already too big.

Also it means that UI-based approach described above must be trc tags dependent (not project wide). So, I think it should be discussed in Bublik project in details.

**Comment by okt-kostik** [comment]
@ol-andrewr 

So you're leaning towards the UI-based approach w/o any changes in the TRC code/files, right?

In this case in UI the result might be NOK, but in the console log it will be green. Are we fine with this? And it's not clear how to match counters, but may be most users should look at the WebUI instead.

A small improvement that might be done later on - TRC might cache bugs classification from bublik in a local file (and include at the bottom in the report). Thus:

 - TRC caches bugs table
 - Run.sh bases on the cache results.

**Comment by ol-andrewr** [comment]
Yes, I'd prefer to not complicate it in TRC. It is just about expectations without classification of these expectations OK/NOK.

**Comment by okt-kostik** [comment]
I was thinking - should we say:

- PASSED
- FAILED, classified
- PASSED, unclassified
- FAILED, unclassified

instead of expected/unexpected in the text version? I.e. if we say that OK/NOK move to UI we don't have expected/unexpected in a sense of "OK/NOK", we have classified failures and unclassified results.

**Comment by ol-andrewr** [comment]
IMHO, we should not complicate TE here. The functionality is required to managers and managers use Bublik to view testing result. So let's add the functionality to UI which has chances to keep it understandable and useable.
